src/entry-client.js

```diff
@@ -79,21 +86,21 @@ export default function initializeClient(createApp, clientOpts) {
 
         // After routing, unregister any dynamic Vuex modules from prior components
         router.afterEach((to, from) => {
-            const priorComponents = router.getMatchedComponents(from);
-
-            priorComponents
+            const fetchDataArgs = { app, route: to, router, store, from };
+            const toModuleNames = router.getMatchedComponents(to).map(c => getModuleName(c, to));
+            router.getMatchedComponents(from)
                 .filter(c => 'vuex' in c)
+                .filter(c => !shouldIgnoreRouteUpdate(c, fetchDataArgs))
```

Comment:
Same logic as above - no need to deregister vuex modules on ignored route updates